src/fe-common/core/fe-common-core.c

```diff
@@ -462,25 +463,59 @@ void fe_common_core_finish_init(void)
 gboolean strarray_find_dest(char **array, const TEXT_DEST_REC *dest)
 {
 	g_return_val_if_fail(array != NULL, FALSE);
-
-	if (strarray_find(array, "*") != -1)
+	const char *type = module_find_id_str("WINDOW ITEM TYPE", dest->window->active->type);
+
+	if ((strarray_find(array, "*") != -1) || // we ignore all targets
+		(g_ascii_strcasecmp(type, "CHANNEL") == 0 && strarray_find(array, "#") != -1) || // we ignore all channels
+		(g_ascii_strcasecmp(type, "QUERY") == 0 && // Is this a query?
+		(g_str_has_prefix("=", dest->target) ? // is it a dcc chat?
+			strarray_find(array, "=") != -1 : // are we ignoring dcc chat?
+			strarray_find(array, "@") != -1)) || // are we ognoring regular queries?
+		(strarray_find(array, dest->target) != -1)) // we ignore all channels with specific name
 		return TRUE;
+	else if (dest->server_tag != NULL) {
+		char *prefix = g_strdup_printf("%s/", dest->server_tag);
+		if (strarray_find_prefix(array, prefix)) {
+			GSList *targets = NULL, *iterator = NULL;
+			gboolean found = FALSE;
+
+			if (type != NULL) {
```

Comment:
Is it possible for `type` to be `NULL`? If so when?
You didn't check for it being `NULL` above.